elastic/apm-server
# Surface config parsing error under EA managed mode

## Motivation/summary



By wrapping reloader errors in a way that EA manager understands (i.e. in a MultiError of UnitError), newRunner initialization errors, including config parsing errors, will be surfaced and logged.

## Checklist



~~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~~ I expect this to go into 8.17 first, then 8.16, but there may be uncertainty around this. We can add changelog before the release.

## How to test these changes


Start apm-server in EA managed mode (e.g. on ESS), pass in an invalid integration policy, e.g. setting `apm-server.rum.response_headers` to a string instead of a list. Check that in fleet UI that EA input unit has failed. Check that an error is logged.

## Related issues


Fixes #14560
